### ISSUE TITLE
"buffer size too small" when querying latest skydns and updated dependencies

### ISSUE THREAD

**Issue by devnev** [miekg/dns#439]
After rebuilding skydns and updating all dependencies, we're seeing an error for large SRV records when querying using miekg/dns.

The error message alternates betwen "buffer size too small" and "overflow unpacking uint32".

client program:
```
package main

import "fmt"
import "github.com/miekg/dns"

const addr = "srv.example.com" // put large SRV addr here

func main() {
	msg := &dns.Msg{}
	msg.SetQuestion(dns.Fqdn(addr), dns.TypeSRV)
	msg.SetEdns0(dns.DefaultMsgSize, false)
	r, err := dns.Exchange(msg, "127.0.0.1:10053")
	fmt.Printf("Sent\n%s\nGot\n%s\nerr:%T:%v\ntruncated:%v\n", msg, r, err, err, err == dns.ErrTruncated)
}
```

output:
```
Sent
;; opcode: QUERY, status: NOERROR, id: 17544
;; flags: rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; QUESTION SECTION:
;srv.example.com.	IN	 SRV

;; ADDITIONAL SECTION:

;; OPT PSEUDOSECTION:
; EDNS: version 0; flags: ; udp: 4096

Got
<nil> MsgHdr
err:*dns.Error:dns: overflow unpacking uint32
truncated:false
```

tcpdump:
```
12:50:49.601519 IP (tos 0x0, ttl 64, id 47671, offset 0, flags [DF], proto UDP (17), length 99)
    127.0.0.1.32931 > 127.0.0.1.10053: [bad udp cksum 0xfe62 -> 0x8f71!] UDP, length 71
12:50:49.625714 IP (tos 0x0, ttl 64, id 47673, offset 0, flags [DF], proto UDP (17), length 4546)
    127.0.0.1.10053 > 127.0.0.1.32931: [bad udp cksum 0x0fc2 -> 0x7d51!] UDP, length 4518
```

**Comment by miekg** [comment]
So... your buffer is 4K, how large is the reply? Or do you mean that SkyDNS should set TC and doesn't? Anycase does not look like a Go dns bug.

**Comment by miekg** [comment]
Might be fixed with #454

**Comment by miekg** [comment]
not a go dns issue.
